dashboard/app/controllers/followers_controller.rb

```diff
@@ -14,6 +14,24 @@ def create
     redirect_to redirect_url, notice: I18n.t('follower.added_teacher', name: @section.teacher.name)
   end
 
+  # As a logged-in student, join a section.  Returns JSON.  Designed to be called by ajax request.
+  def create_async
+    @section = Section.find_by_code params[:section_code]
+    if @section
+      @section.add_student current_user
+      render json: {
+        result: "success",
+        sections: current_user.sections_as_student.map(&:summarize),
+        notice: I18n.t('follower.added_teacher', name: @section.teacher.name)
+      }
+    else
+      render json: {
+        result: "error",
+        sections: current_user.sections_as_student.map(&:summarize)
+      }
```

Comment:
This would happen automatically if we used CanCan's methods, e.g. `load_resource`, or even if we use the more aggressive bang find method `find_by!(code: params[:section_code])`. In either case, it will raise an `ActiveRecord::RecordNotFound` which will cause Rails to send a 404 response.